lib/improver/cube_combiner.py

```diff
@@ -47,72 +47,274 @@ class CubeCombiner(object):
+        new_coord = new_coord_method(long_name=coord_name,
+                                     points=points,
+                                     bounds=bounds,
+                                     units=units)
+        result.add_aux_coord(new_coord)
+        if metatype == 'DimCoord':
+            result = iris.util.new_axis(result, coord_name)
+        if self.warnings_on:
+            msg = ("Adding new coordinate "
+                   "{} with {}".format(coord_name,
+                                       changes))
+            warnings.warn(msg)
+        return result
+
+    def update_coord(self, cube, coord_name, changes):
+        """Amend the metadata in the combined cube.
+
+        Args:
+            cube (iris.cube.Cube):
+                Cube containing combined data.
+            coord_name (string):
+                Name of the coordinate being updated.
+            changes (string or dict):
+                Details on coordinate to be updated.
+                If changes = 'delete' the coordinate is deleted.
+
+        Returns:
+            result (iris.cube.Cube):
+                Cube with updated coordinate.
+
+        Raises:
+            ValueError : Can only remove a coordinate of length 1
+            ValueError : Mismatch in points in existing coord
+                and updated metadata.
+            ValueError : Mismatch in bounds in existing coord
+                and updated metadata.
+            ValueError : The shape of the bounds array should
+                be points.shape + (n_bounds,)
+            UserWarning: Deleted coordinate.
+            UserWarning: Updated coordinate
+
+        """
+        new_coord = cube.coord(coord_name)
+        result = cube
+        if changes == 'delete':
+            if len(new_coord.points) != 1:
+                msg = ("Can only remove a coordinate of length 1"
+                       " coord  = {}".format(coord_name))
+                raise ValueError(msg)
+            result.remove_coord(coord_name)
+            result = iris.util.squeeze(result)
+            if self.warnings_on:
+                msg = ("Deleted coordinate "
+                       "{}".format(coord_name))
+                warnings.warn(msg)
+        else:
+            if 'points' in changes:
+                new_points = np.array(changes['points'])
+                if (len(new_points) ==
+                        len(new_coord.points)):
+                    new_coord.points = new_points
+                else:
+                    msg = ("Mismatch in points in existing"
+                           " coord and updated metadata for "
+                           " coord {}".format(coord_name))
+                    raise ValueError(msg)
+            if 'bounds' in changes:
+                new_bounds = np.array(changes['bounds'])
+                if new_coord.bounds is not None:
+                    if (len(new_bounds) == len(new_coord.bounds) and
+                            len(new_coord.points)*2 ==
```

Comment:
Similarly to the check on the points above, I think this could be reduced to just `new_coord.bounds = new_bounds`, as an exception will already be raised by Iris.

Reply:
As above, we might be changing several coordinates. Is the Iris error descriptive enough?